src/files.cpp

```diff
@@ -0,0 +1,89 @@
+#include <fstream>
+#include <sstream>
+#include "files.h"
+
+
+FileManager::Exception::Exception(std::string msg) : msg{std::move(msg)} {}
```

Comment:
As far as I remember, `std::move` is obsolete here, as it is just a static cast to an rvalue. And in this particular example, it will be detrimental to the compiler as it will prevent it from performing copy-elision. So I'd just chuck the call to `std::move` away. 😺 

```suggestion
FileManager::Exception::Exception(std::string msg) : msg{msg} {}
```

Reply:
I can't find any information, that copy-ellision can happen here.
As far as I know move is necessary.

Reply:
@nieznanysprawiciel You're right, it doesn't happen when passing in the arguments to the constructor. My bad, it should use `std::move` as it was originally.